sycl/doc/extensions/LevelZeroBackend/LevelZeroBackend.md

```diff
@@ -0,0 +1,119 @@
+﻿# Level-Zero backend specification
+
+## 1. Introduction
+
+This extension introduces Level-Zero backend for SYCL.
+It is built on top of Level-Zero runtime enabled with [Level-Zero API](https://spec.oneapi.com/level-zero/latest/index.html).
+The Level-Zero backend is aimed to provide the best possible performance of SYCL application on a variety of targets supported.
+The currently suppported targets are all Intel GPUs starting with Gen9.
+
+NOTE: This specification is a draft. While describing the currently implemented behaviors it is known to be not complete nor exhaustive.
+      We shall continue to add more information, e.g. explain general mapping of SYCL programming model to Level-Zero API.
+      It will also be gradually changing to a SYCL-2020 conforming implementation.
+
+## 2. Prerequisites
+
+The Level-Zero loader and drivers need to be installed on the system for SYCL runtime to recognize and enable the Level-Zero backend.
+For further details see <https://software.intel.com/content/www/us/en/develop/articles/intel-oneapi-dpcpp-system-requirements.html>.
+                  
+## 3. User-visible Level-Zero backend selection (and the default backend)
+
+The Level-Zero backend is added to the cl::sycl::backend enumeration:
+
+``` C++
+enum class backend {
+  // ...
+  level_zero,
+  // ...
+};
+```
+
+There are multiple ways in which the Level-Zero backend can be selected by the user.
+        
+### 3.1 Through an environment variable
+        
+The SYCL_DEVICE_FILTER environment variable limits the SYCL runtime to use only a subset of the system's devices.
+By using ```level_zero``` for backend in SYCL_DEVICE_FILTER you can select the use of Level-Zero as a SYCL backend.
+For further details see here: <https://github.com/intel/llvm/blob/sycl/sycl/doc/EnvironmentVariables.md>.
+        
+### 3.2 Through a programming API
+        
+There is an extension that introduces a filtering device selection to SYCL described in
+<https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/FilterSelector/FilterSelector.adoc>.
+Similar to how SYCL_DEVICE_FILTER applies filtering to the entire process this device selector can be used to
+programmatically select the Level-Zero backend.
+                
+When neither the environment variable nor the filtering device selector are used, the implementation chooses
+the Level-Zero backend for GPU devices supported by the installed Level-Zero runtime.
+The serving backend for a SYCL platform can be queried with the ```get_backend()``` member function of ```cl::sycl::platform```.
+
+## 4. Interoperability with the Level-Zero API
+
+This chapter describes the various interoperability possible between SYCL and Level-Zero.
+The application must include both of the following headers in order to use any of the interoperation APIs described in this section,
+and they must be included in the order shown:

```

Comment:
Could you please clarify why order matters? Is it possible to relax this restriction?

Reply:
Because currently the "sycl/backend/level_zero.hpp" does not include "level_zero/ze_api.h" but relies on the thing defined there. There is an ongoing discussion if we'd able to relax this and just include the "level_zero/ze_api.h" from the "sycl/backend/level_zero.hpp"